azavea/raster-foundry
# S3 upload process

## Overview

This PR adds backend handling to uploads of the 'S3' type. It takes advantage of the fact that our upload process is essentially operating from S3 as it is now.

There is now a `source` property on uploads which for S3 uploads is the bucket the imagery is in. The `api/uploads` endpoint handles populating `upload.files` with all geotiffs in that bucket, and then kicks of the import as usual.

There is definitely some cleanup to do but it works. I think this is the most sensible way to handle S3 imports as it makes no changes to the import process itself.

### Checklist

~- [ ] Styleguide updated, if necessary~

### Notes

This is based on #2201 for testing purposes

## Testing Instructions

 * Start up the server with airflow (make sure you've run migrations)
 * Find a bucket with some geotiffs in it that's accessible to the backend and formulate the 's3://...' path for that bucket
 * From the front-end, start an import, select S3 and enter the bucket, submit
 * Watch it go

Closes #2146 
Closes #2147
